components/org.wso2.carbon.identity.data.publisher.authentication.analytics.login/src/main/java/org/wso2/carbon/identity/data/publisher/authentication/analytics/login/AnalyticsLoginDataPublisherUtils.java

```diff
@@ -185,8 +189,16 @@ public static AuthenticationData buildAuthnDataForAuthentication(Event event) {
         authenticationData.setEventType(AnalyticsLoginDataPublishConstants.OVERALL_EVENT);
         authenticationData.setContextId(context.getContextIdentifier());
         authenticationData.setEventId(UUID.randomUUID().toString());
-        authenticationData.setAuthnSuccess(true);
-        authenticationData.setRemoteIp(IdentityUtil.getClientIpAddress(request));
+        if (status.equals(AuthenticatorStatus.PASS)) {
+            authenticationData.setAuthnSuccess(true);
+        } else {
```

Comment:
AuthenticatorStatus can be PASS, FAIL or CONTINUE. So shall we handle this as an if else block for FAIL status?

Reply:
Modified.